Framework/PythonInterface/core/inc/MantidPythonInterface/core/PropertyWithValueExporter.h

```diff
@@ -43,7 +43,8 @@ struct PropertyWithValueExporter {
     using namespace Mantid::Kernel;
 
     class_<PropertyWithValue<HeldType>, bases<Property>, boost::noncopyable>(
-        pythonClassName, init<std::string, HeldType>((arg("self"), arg("name"), arg("value"))))
+        pythonClassName, init<std::string, HeldType, unsigned int>(
```

Comment:
Don't trick the compiler with auto-casting `enum` to `int`. [Mantid::Kernel::Direction](https://github.com/mantidproject/mantid/blob/main/Framework/Kernel/inc/MantidKernel/Property.h#L50-L57) is exposed through python already.

Also, good use of default values.

Reply:
I put this because the compiler said it wanted an `unsigned int` (which surprised me because I thought these defaulted to `char`), but lemme see if it will work with the enum

Reply:
Nevermind. The function definitions appear to be unsigned integers. Example
```c++
  template <typename T>
  void declareProperty(const std::string &name, T value, IValidator_sptr validator = std::make_shared<NullValidator>(),
                       const std::string &doc = "", const unsigned int direction = Direction::Input) {
    std::unique_ptr<PropertyWithValue<T>> p = std::make_unique<PropertyWithValue<T>>(name, value, validator, direction);
    declareProperty(std::move(p), doc);
  }

...

  template <typename T>
  void declareProperty(const std::string &name, T value, const std::string &doc,
                       const unsigned int direction = Direction::Input) {
    std::unique_ptr<PropertyWithValue<T>> p =
        std::make_unique<PropertyWithValue<T>>(name, value, std::make_shared<NullValidator>(), direction);
    declareProperty(std::move(p), doc);
  }
```

Reply:
It is actually `PropertyWithValue` that is setup to want an `unsigned int` as the direction.  I can change it, but then it breaks some other stuff

Reply:
I missed your reply before commenting